components/script/dom/htmlinputelement.rs

```diff
@@ -1193,7 +1193,47 @@ impl HTMLInputElement {
             },
             // https://html.spec.whatwg.org/multipage/#range-state-(type=range):value-sanitization-algorithm
             InputType::Range => {
-                value.set_best_representation_of_the_floating_point_number();
+                let minimum = self.Min().parse().unwrap_or(0f64);
```

Comment:
Would be nice to not directly go through the IDL getters to avoid a `DOMString` copy. AFAIK `min`, `max` and `step` attribute values should be stored as integers in `HTMLInputElement`, feel free to ping me on IRC if you need more details about that.